pkg/datasource/getInsights.go

```diff
@@ -0,0 +1,84 @@
+package datasource
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/grafana/grafana-plugin-sdk-go/backend"
+	"github.com/grafana/grafana-plugin-sdk-go/backend/log"
+	"github.com/grafana/grafana-plugin-sdk-go/data"
+	xray "github.com/grafana/x-ray-datasource/pkg/xray"
+)
+
+type GetInsightsQueryData struct {
+	Query string `json:"query"`
+}
+
+func (ds *Datasource) getInsights(ctx context.Context, req *backend.QueryDataRequest) (*backend.QueryDataResponse, error) {
+	xrayClient, err := ds.xrayClientFactory(&req.PluginContext)
+	if err != nil {
+		return nil, err
+	}
+
+	response := &backend.QueryDataResponse{
+		Responses: make(map[string]backend.DataResponse),
+	}
+
+	for _, query := range req.Queries {
+		response.Responses[query.RefID] = getSingleInsight(xrayClient, query)
+	}
+
+	return response, nil
+}
+
+func getSingleInsight(xrayClient XrayClient, query backend.DataQuery) backend.DataResponse {
+	insightsResponse, err := xrayClient.GetInsightSummaries(&xray.GetInsightSummariesInput{
```

Comment:
Isn't there a `GetInsightSummariesPages` api? If so we should probably use that so we can limit the number of results if needed.

Reply:
There isn't. But there is a GetInsightSummariesWithContext. Should that be used instead?

Reply:
Probably no need right now. Context makes sense in cases you are doing async stuff with go routines and you can pass different timeouts or link go routines that should be ended together in a group but probably no need for that atm.